src/muc_light/mod_muc_light_db_rdbms.erl

```diff
@@ -774,28 +773,27 @@ modify_aff_users_transaction(MainHost, RoomUS, RoomID, AffUsersChanges,
             Error
     end.
 
--spec apply_aff_users_transaction(MainHost :: jid:lserver(),
+-spec apply_aff_users_transaction(HostType :: mongooseim:host_type(),
                                   RoomID :: room_id(),
                                   AffUsersChanges :: aff_users(),
                                   JoiningUsers :: [jid:simple_bare_jid()]) -> ok.
-apply_aff_users_transaction(MainHost, RoomID, AffUsersChanged, JoiningUsers) ->
+apply_aff_users_transaction(HostType, RoomID, AffUsersChanged, JoiningUsers) ->
     lists:foreach(
       fun({{UserU, UserS}, none}) ->
-              {updated, _} = delete_aff(MainHost, RoomID, UserU, UserS);
+              {updated, _} = delete_aff(HostType, RoomID, UserU, UserS);
          ({{UserU, UserS} = UserUS, Aff}) ->
               case lists:member(UserUS, JoiningUsers) of
                   true ->
-                      {updated, _} = insert_aff(MainHost, RoomID, UserU, UserS, Aff);
+                      {updated, _} = insert_aff(HostType, RoomID, UserU, UserS, Aff);
                   false ->
-                      {updated, _} = update_aff(MainHost, RoomID, UserU, UserS, Aff)
+                      {updated, _} = update_aff(HostType, RoomID, UserU, UserS, Aff)
               end
       end, AffUsersChanged).
 
 %% ------------------------ Common ------------------------
 
--spec main_host(JIDOrServer :: jid:simple_bare_jid() | binary()) -> jid:lserver().
+-spec main_host(JIDOrServer :: jid:simple_bare_jid() | binary()) -> mongooseim:host_type().
 main_host({_, RoomS}) ->
     main_host(RoomS);
 main_host(MUCServer) ->
-    {ok, MainHost} = mongoose_subhosts:get_host(MUCServer),
-    MainHost.
+    mod_muc_light_utils:muc_host_to_host_type(MUCServer).
```

Comment:
I think that this function can change its name, it doesn't really reflect anymore what it used to do.